crypto/rand/rand_win.c

```diff
@@ -34,52 +39,57 @@
 #  define INTEL_DEF_PROV L"Intel Hardware Cryptographic Service Provider"
 # endif
 
-static void readtimer(void);
-
 int RAND_poll(void)
 {
-    MEMORYSTATUS mst;
-# ifndef RAND_WINDOWS_USE_BCRYPT
+# ifndef USE_BCRYPTGENRANDOM
     HCRYPTPROV hProvider;
 # endif
     DWORD w;
-    BYTE buf[64];
-
-# ifdef RAND_WINDOWS_USE_BCRYPT
-    if (BCryptGenRandom(NULL, buf, (ULONG)sizeof(buf), BCRYPT_USE_SYSTEM_PREFERRED_RNG) == STATUS_SUCCESS) {
-        RAND_add(buf, sizeof(buf), sizeof(buf));
+    BYTE buf[RANDOMNESS_NEEDED];
+    int ok = 0;
+
+# ifdef OPENSSL_RAND_SEED_RDTSC
+    {
+        unsigned char c;
+        int i;
+
+        for (i = 0; i < 10; i++) {
+            /* Low-order bits of the high-speed clock. */
+            c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
+            RAND_add(&c, 1, 0.5);
```

Comment:
Should the rdtsc stuff get factored out into a helper function (it appears in the same form in two places)?